types/object.go

```diff
@@ -52,6 +52,10 @@ func (o ObjectType) ValueFromTerraform(ctx context.Context, in tftypes.Value) (a
 	object := Object{
 		AttrTypes: o.AttrTypes,
 	}
+	if in.Type() == nil {
+		object.Null = true
+		return object, nil
```

Comment:
This could possibly also return `nil, nil`, but I wasn't sure if everything that called `ValueFromTerraform` was prepared to check for a `nil` value if the error was also `nil`.

It seemed reasonable that if converting a value with no type to a value with a type the result would be a null value with the target type, but I could see arguments for other conventions too.

Reply:
I think returning the correct type with a null value is probably best here. 👍